Add option to leave DOM in place when destroy() is called, add proper cleanup of generated_stylesheets.

billschaller opened the pull request:
Add the option suggested in #218 to allow clean destroy by running destroy(false). This allows you to use gridster  on an element, and then destroy gridster without destroying the original DOM elements. This is what I'm using to cleanly implement window resizing at the moment.

Add some code to properly strip out generated style tags when destroying the gridster instance. It also removes the entries in the Gridster.generated_stylesheets static var. This should allow multiple gridster instances to play more nicely on the same page. This should take care of #284, #227, and is a more complete solution than #287.

dustmoo committed:
Added grid-swap behavior

dustmoo committed:
Added property to prevent larger widget shifting

dustmoo committed:
Updated Readme

dustmoo committed:
More predictable default behavior

dustmoo committed:
Added swaping queue, better handling of large to small swaps

dustmoo committed:
Added swaping queue, better handling of large to small swaps

dustmoo committed:
working on map grid functionality

dustmoo committed:
Fix row issue

dustmoo committed:
Swap queue awesomeness

dustmoo committed:
Swap queue awesomeness

dustmoo committed:
Good enough swap logic for now

dustmoo committed:
Removed console.logs

dustmoo committed:
Removed console.logs

dustmoo committed:
Added max_rows and drag restriction - ready for integration :)

dustmoo committed:
Serialize bug fix

dustmoo committed:
Version Updates

dustmoo committed:
Merge branch 'mapapp'

dustmoo committed:
Updated Readme

dustmoo committed:
Updated Readme

dustmoo committed:
Working on Static widgets

sbehrends committed:
Added disabled dragging init feature gridster.js

In initial options you can disble draggin by adding disable_drag: true on options list at initialising gridster

dustmoo committed:
Merge pull request #7 from sbehrends/patch-1

Added disabled dragging init feature gridster.js - Thank you sbehrends

dustmoo committed:
Added Swap Demo

dustmoo committed:
Merge branch 'master' of https://github.com/dustmoo/gridster.js

dustmoo committed:
rebuilt dist

dustmoo committed:
update readme

dustmoo committed:
Better handling of widgets slideing down

dustmoo committed:
Updated sample

dustmoo committed:
Update Readme

Commit:
fix(destroy): Added bool param to destroy method to leave original DOM in place

Added a new parameter 'removeDOM' to the destroy method - when false,
destroy leaves the DOM in place, but just removes all the gridster
business from it and removes classes and stylesheets it added.

Closes #218

Commit:
fix(destroy): Added better cleanup for style tag cache when destroy() is called.

The static stylesheet cache generated_stylesheets is now cleaned up
properly when a gridster instance is destroyed. Adds code to properly strip out generated style tags when destroying the gridster instance. It also removes the entries in the Gridster.generated_stylesheets static var. This should allow multiple gridster instances to play more nicely on the same page.

Closes #227, #284, #287

Commit:
fix(destroy) fixed styles removal in ie

ciaranj commented:
:+1:  This patch is spot on for adding + removing gridster from the same page repeatedly :)

hrosenbauer committed:
add resize_widget_dimension()
resize widgets after initialization

hrosenbauer committed:
add resize_widget_dimension()
resize widgets after initialization

Conflicts:
	src/jquery.gridster.js

andreas-trad committed:
Prevent multi <style> generation 

If the autogenerate_stylesheet option is set to true then the library is adding multiple times the same style on the head of the document (see the source of the http://gridster.net/demos/adding-widgets-dynamically.html demo in order to observe this). With this correction the gridster style element is only inserted once on the style.

kwak27 committed:
fix add_widget to use correct size_y when adding rows

hrosenbauer committed:
fix min-width and min-height after resize

remove min-width and min-height after stopping resize

hrosenbauer committed:
fix min-width and min-height after resize

remove min-width and min-height after stopping resize

Commit:
fixes-add-widget-overlapping-issue-#340

dasmall committed:
Removing previously added style tags before adding new one.
Hopefully fixes #211 and #294.

Commit:
Fixed issue where existing widget position would be overwritten on gridmap when adding a faux cell to accommodate another widget.

Commit:
Fixed typeof comparison to match against string.

hrosenbauer committed:
Merge branch 'resize-widget-dimensions' of https://github.com/rebugger/gridster.js into resize-widget-dimensions

Conflicts:
	src/jquery.gridster.js

AWhetter committed:
docs(gridster): fixed a typo in Gridster class documentation

AWhetter committed:
Fixed a typo in Gridster class documentation

dragonier committed:
add_resize_handle: multiple span (resizeable) entries in widgets

On init() or get_widgets_from_DOM() new span entries are added to the
widgets container.

Commit:
Create bower.json

Commit:
fix(gridster): changed "instanceof jQuery" to "instanceof $"

Original code will break if jquery noconflict is applied, where $ and jQuery produces jQuery instance of different versions.

Commit:
Changed "instanceof jQuery" to "instanceof $"

Original code will break if jquery noconflict is applied, where $ and jQuery produces jQuery instance of different versions.

rept committed:
Wrong addition solved in add_faux_rows by adding parseInt

rept committed:
fix(gridster): wrong addition solved in add_faux_rows/cols by adding parseInt

Closes #426, #425

rept committed:
Also added parseInt to the add_faux_cols method

rept committed:
Also added parseInt to the add_faux_cols method

joaoafrmartins committed:
feat(gridster): Common.js support

Closes #434

joaoafrmartins committed:
support UMD

joaoafrmartins committed:
commonjs

yfix committed:
Merge pull request #1 from ducksboard/master

Update from origin

Commit:
Fix gridster destroy

Removes the coords data from each element.

Commit:
Update jquery.gridster.js

Commit:
Added ability to specify scroll container for drag/resize

Commit:
fix issue #469

remove trailing comma to be more IE compliant

Commit:
fix (Gridster.js):  destroy  resize_api

Call  resize_api.destroy()  to fix  memory leak.

Commit:
fix(gridster): destroy resize_api

Call  `resize_api.destroy()`  to fix  memory leak.

Closes #473

Commit:
Added destroy of widget coords and resize api

sogawa committed:
rename test files.

sogawa committed:
add element "#qunit-testresult"

ericmann committed:
Accepting Resize API pull request. @props tormoalto

sogawa committed:
Convert to integer type of property value when sorting.

ericmann committed:
Cache $(window) and $(this) selectors for performance

ericmann committed:
Fix the memory leak that was incorrectly iterating rows and columns

Commit:
add model_id parameter to add_widget method in order to create a relationship between widget and backbone model

Commit:
re-name model_id parameter

veeamGMBH committed:
delete jquery dependency

Commit:
Fix spelling mistake

ericmann committed:
Fix Widget Right Edge Collission Detection

Apply the fix from duckboard's issue tracker. Fixes #2.

ericmann committed:
Grunt

ericmann committed:
Bumping to v0.5.7

Commit:
fix(gridster): changed the way widgets were getting positioned so that margins are actually the same size that the user specified all the way around the grid

feat(gridster): added responsive nature to grid

Commit:
Release v0.6.0

Commit:
fix(gridster): fixing drag limit issues when using autogrow_cols

Commit:
fix(gridster): fixed expand_widget bug not expanding full width of window
fix(gridster): user can specify size_x when expanding widget

Commit:
Release v0.6.1

Commit:
fix(gridster): forcing height of gridster container to auto when in collapsed mode

Commit:
Release v0.6.2

Commit:
fix(gridster): fixes bug where widgets would overlay other widgets after a resize

fhawkes committed:
Merge pull request #2 from DecksterTeam/resize-overlap

fix(gridster): fixes bug where widgets would overlay other widgets after...

Commit:
fix(gridster): adding bower.json file

Commit:
Merge branch 'master' of github.com:DecksterTeam/gridster.js

Commit:
fix(gridster): adding bower.json file

Commit:
Fix for bug #493 including unit tests to prove.

Run npm install and visit test/jquery.gridster-amd.html in browser.

Commit:
refactor(gridster): adding callback to add_widget function

Commit:
fix(gridster): fixing resize limits when in fixed width mode
feature(gridster): added fix_to_content function to fit widget dimensions to a given width and height
feature(gridster): added resize_widget_dimensions function
refactor(gridster): moved code to set number of columns in grid to helper function

Commit:
Release v0.6.3

ericmann committed:
Add an optional final parameter to add_widget.

ericmann committed:
Grunt changes

ericmann committed:
Merge in fade changes

diogobenica committed:
Change jQuery reference for $.

Commit:
Include move_widget function (Move a widget to a specific row and column.).

Commit:
add function move_widget in dist/jquery.gridster.js

Commit:
#3 Adds ability to automatically center widgets in the grid.

Commit:
Release v0.6.4

Commit:
#3 Adds ability to automatically center widgets in the grid.

Commit:
Move the check for responsive width above the calculations of width.  This will prevent the calculated values changing the width for browsers which don't respect min-width

Commit:
Move the check for responsive width above the calculations of width.  This will prevent the calculated values changing the width for browsers which don't respect min-width

Commit:
fix(gridster): fixed bugs in centering_widgets (widgets were getting smushed when being resized) and fixed bug with min_width

Commit:
Release v0.6.5

Commit:
Release v0.6.5

Commit:
chore(grunt): fixing versioning system

Commit:
Merge branch 'master' of https://github.com/DecksterTeam/gridster.js

Commit:
fix a NAN error that could occur if using center_widget if the grid was in responsive mode.
Error check the shift cols, to ensure they can't move left past col 1
make a change from 100% to 100VW for better browser support of min-width

Commit:
Merge from abril/gridster.js

* merging:
  add function move_widget in dist/jquery.gridster.js
  Include move_widget function (Move a widget to a specific row and column.).
  Change jQuery reference for $.

Commit:
Merge remote-tracking branch 'eric/master'

  Grunt changes
  Add an optional final parameter to add_widget.
  Bumping to v0.5.7
  Grunt
  Fix Widget Right Edge Collission Detection
  re-name model_id parameter
  add model_id parameter to add_widget method in order to create a relationship between widget and backbone model
  Fix the memory leak that was incorrectly iterating rows and columns
  Cache $(window) and $(this) selectors for performance
  fix (Gridster.js):  destroy  resize_api

Commit:
Merge remote-tracking branch 'ducksboard/bug-493-jquery-no-global-fail'

* ducksboard/bug-493-jquery-no-global-fail:
  Fix for bug #493 including unit tests to prove.

Commit:
Merge remote-tracking branch 'Roger-Waters/gridster.js'

*   Fix spelling mistake

Commit:
Merge remote-tracking branch 'sogawa/gridster.js/strict-sort'

* sogawa/gridster.js/strict-sort:
  Convert to integer type of property value when sorting.
  add element "#qunit-testresult"
  rename test files.

Commit:
Merge remote-tracking branch 'bfarka/gridster.js/fix-trailing-comma'

* bfarka/gridster.js/fix-trailing-comma:
  fix issue #469

Commit:
Merge remote-tracking branch 'mooglemoogle/gridster.js/master'

* mooglemoogle/gridster.js/master:
  Added destroy of widget coords and resize api
  Added ability to specify scroll container for drag/resize

Commit:
Merge remote-tracking branch 'Toosick/gridster.js/patch-1'

* Toosick/gridster.js/patch-1:
  Update jquery.gridster.js
  Fix gridster destroy

Commit:
Merge remote-tracking branch 'joaoafrmartins/gridster.js/master'

* joaoafrmartins/gridster.js/master:
  commonjs
  support UMD

Commit:
Merge remote-tracking branch 'rept/gridster.js/addition_problem_add_faux_rows'

* rept/gridster.js/addition_problem_add_faux_rows:
  Also added parseInt to the add_faux_cols method
  Also added parseInt to the add_faux_cols method
  Wrong addition solved in add_faux_rows by adding parseInt

Commit:
Merge remote-tracking branch 'aegisrunestone/gridster.js/master'

* aegisrunestone/gridster.js/master:
  Changed "instanceof jQuery" to "instanceof $"

Commit:
Merge remote-tracking branch 'Tropicalista/gridster.js/master'

Commit:
Merge remote-tracking branch 'dragonier/gridster.js/master'

* dragonier/gridster.js/master:
  add_resize_handle: multiple span (resizeable) entries in widgets

Commit:
Merge remote-tracking branch 'AWhetter/gridster.js/fix_doc_typo'

* AWhetter/gridster.js/fix_doc_typo:
  Fixed a typo in Gridster class documentation

Commit:
Merge remote-tracking branch 'cpascu/gridster.js/faux-rows-patch'

* cpascu/gridster.js/faux-rows-patch:
  Fixed typeof comparison to match against string.
  Fixed issue where existing widget position would be overwritten on gridmap when adding a faux cell to accommodate another widget.

Commit:
Merge remote-tracking branch 'dasmall/gridster.js.git/master'

* dasmall/gridster.js.git/master:
  Removing previously added style tags before adding new one. Hopefully fixes #211 and #294.

Commit:
Merge remote-tracking branch 'AdrianTeh/gridster.js.git/master'

* AdrianTeh/gridster.js.git/master:
  fixes-add-widget-overlapping-issue-#340

Commit:
Merge remote-tracking branch 'kwak27/gridster.js/fix_add_widget'

* kwak27/gridster.js/fix_add_widget:
  fix add_widget to use correct size_y when adding rows

Commit:
Merge remote-tracking branch 'andreas-trad/gridster.js.git/patch-1'

* andreas-trad/gridster.js.git/patch-1:
  Prevent multi <style> generation

Commit:
Merge remote-tracking branch 'rebugger/gridster.js.git/resize-widget-dimensions'

* rebugger/gridster.js.git/resize-widget-dimensions:
  fix min-width and min-height after resize
  add resize_widget_dimension() resize widgets after initialization
  fix min-width and min-height after resize
  add resize_widget_dimension() resize widgets after initialization

Commit:
Merge remote-tracking branch 'yfix/gridster.js.git/master'

* yfix/gridster.js.git/master:

Commit:
Merge remote-tracking branch 'Laboratory/gridster.js.git/master'

* Laboratory/gridster.js.git/master:
  delete jquery dependency

Commit:
Update Readme

Commit:
Release v0.6.6

Commit:
Fix my merge error from the merge from remote-tracking branch 'cpascu/gridster.js/faux-rows-patch'

Commit:
Merge from dustmoo/gridster.js

* commit '262d115cf8b5bdaed40a7e5b6aaa6df6e71b6a07':
  Added swaping queue, better handling of large to small swaps
  Added swaping queue, better handling of large to small swaps
  More predictable default behavior
  Updated Readme
  Added property to prevent larger widget shifting
  Added grid-swap behavior

Commit:
Merge from dustmoo/gridster.js

* commit '8f431a423302c7c58f511d5ede9270077993f9b3':
  Updated Readme
  Updated Readme
  Version Updates
  Serialize bug fix
  Added max_rows and drag restriction - ready for integration :)
  Removed console.logs
  Removed console.logs
  Good enough swap logic for now
  Swap queue awesomeness
  Swap queue awesomeness
  Fix row issue
  working on map grid functionality

Commit:
Merge commit '6eb521015cda5212f6fa92492dbe4050bd92e00c'

* commit '6eb521015cda5212f6fa92492dbe4050bd92e00c':
  Update Readme
  Updated sample
  Better handling of widgets slideing down
  update readme
  rebuilt dist
  Added Swap Demo
  Added disabled dragging init feature gridster.js
  Working on Static widgets

Commit:
Update Samples

Commit:
bring the Demo's into  master for testing & Automate the updates to gh-pages branch

Commit:
Add unit testing for core API, ensure all demos work, and add unit testing to ensure they stay working.

vieron committed:
feat(draggable): autoscrolling

If the element being dragged is inside a scrollable container, keeping the
mouse in one of the sides of the container will trigger scroll.

vieron committed:
fix(gridster): ensure widget dimensions and coords are always ints

Thanks to @sogawa for reporting

vieron committed:
chore: bump up jquery version to 2.1.3

vieron committed:
feat(gridster): browserify compatibility

Commit:
fix(gridster): preventing gridster from adding extra resize handles

Commit:
Revert "fix(gridster): preventing gridster from adding extra resize handles"

This reverts commit 9d077da676826606243c2552dc9997c492687203.

Commit:
preventing gridster from creating extra resize handles while centering

fhawkes committed:
Merge pull request #6 from DecksterTeam/dadams

Resize handles

Commit:
Release v0.6.6

vieron committed:
tests(gridster): add basic test suite for gridster

Tests are build with mocha + chai, and you can run them from the command line with `grunt karma` or opening `test/index.html` in your browser.

This is a first approach, there are hundreds of tests more that should be added.
I’ve ported some tests on PRs but more needs to be done.

vieron committed:
chore(gridster): remove unused code

vieron committed:
feat(gridster.css): remove possible default pading

vieron committed:
feat(gridster): add config to set custom show/hide widget methods

by default jQuery’s `hide` and `show` methods are used. You could also
use fadeIn/fadeOut or write your own kind of jQuery plugin like
`$.fn.showInAFancyWay` and use `showInAFancyWay` as the value in the
show_method config option.

If you want to keep the previos behaviour, you need to set
`hide_method` option to `’fadeOut’`

Breaking Changes

`remove_widget` and `remove_all_widgets` methods not return a promise
instead of the gridster instance

vieron committed:
fix(gridster): `resize_widget` also accepts HTMLElements

for consistency with the `remove_widget`

vieron committed:
fix(gridster): ensure coords instances are destroyed on widgets

…when removing a widget, or when destroying the gridster instance.
Before this change, if you called destroy without `remove` set to true,
coords instances was leaking jQuery data objects.

Also nullified the gridmap, and faux_grid arrays.

vieron committed:
style(jshint): add jshint config and setup grunt task

vieron committed:
style(jshint): adapt code to pass jshint

Commit:
rev jquery version

Commit:
0.6.7 release

fhawkes committed:
fix(gridster): responsive width now resizes based off wrapper not window

Commit:
update dependencies and jshint

Commit:
Merge commit 'e69c3e8f64aa4557ef032e4d0d8185e83b1aed21'
fix(gridster): responsive width now resizes based off wrapper not window

Commit:
Release v0.6.8

ericop committed:
typo on "the" in "inactivity over the last year"

Commit:
Merge pull request #4 from ericop/patch-1

typo on "the" in "inactivity over the last year"

Commit:
Fix bug with responsive layout if grid in wrapped in a container which has padding. It now limits the width of the widget to the containing element (also helps with scrollbars)

Commit:
Add new demo page and allow two new features based on Issue #6:
- delay moving components until drag is complete
- allow exact position of grid components (do not move up)

Commit:
Remove debug code

Commit:
Merge remote-tracking branch 'remotes/dsmorse/master'

Conflicts:
	src/jquery.gridster.js
